From 63135e73fd878cb71b1eebf8e877c4d4c34feba7 Mon Sep 17 00:00:00 2001
From: Andreas Jaggi <andreas.jaggi@waterwave.ch>
Date: Fri, 21 Feb 2020 07:40:36 +0100
Subject: ulogd: printpkt: always print IPv6 protocol

Print the protocol number for protocols not known by name.

Signed-off-by: Andreas Jaggi <andreas.jaggi@waterwave.ch>
Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
---
 util/printpkt.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/util/printpkt.c b/util/printpkt.c
index 69a47ca..2aacddb 100644
--- a/util/printpkt.c
+++ b/util/printpkt.c
@@ -355,6 +355,9 @@ static int printpkt_ipv6(struct ulogd_key *res, char *buf)
 			break;
 		}
 		break;
+	default:
+		buf_cur += sprintf(buf_cur, "PROTO=%u ",
+				   ikey_get_u8(&res[KEY_IP6_NEXTHDR]));
 	}
 
 	return buf_cur - buf;
-- 
cgit v1.2.3
